pkg/operator/configobserver/cloudprovider/observe_cloud_volume_plugin.go

```diff
@@ -0,0 +1,96 @@
+package cloudprovider
+
+import (
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/library-go/pkg/cloudprovider"
+	"github.com/openshift/library-go/pkg/operator/configobserver"
+	"github.com/openshift/library-go/pkg/operator/events"
+)
+
+// NewCloudVolumePluginObserver returns a new cloudprovider observer for syncing cloud provider specific
+// information to controller-manager, leaving only storage controller loops enabled.
+func NewCloudVolumePluginObserver(targetNamespaceName string, cloudVolumePluginPath, cloudProviderConfigPath []string) configobserver.ObserveConfigFunc {
```

Comment:
There may be a misunderstanding of what happens when you return an error.  When you return an error, your returned configuration *is still used*.  This is because you're one of a dozen config loops and we cannot halt all configuration on a single observation failure.  This means that you must return the best possible config you can.  Usually, this is the previous configuration value you have.

Reply:
if returning a configuration that is basically the same as the previous, is it best practice to make a copy and return the copy?